include/deal.II/lac/block_vector_base.h

```diff
@@ -2264,6 +2262,17 @@ void BlockVectorBase<VectorType>::extract_subvector_to (ForwardIterator
     }
 }
 
+
+template <typename VectorType>
+inline
+bool BlockVectorBase<VectorType>::supports_distributed_data() const
+{
+  // Create a dummy vector of type VectorType to check
+  // supports_distributed_data even if BlockVectorBase contains no block
+  VectorType dummy;
+  return dummy.supports_distributed_data();
```

Comment:
OK, I do not want to stall this PR. My fundamental issue - or, more precisely, the thing I don't fully understand is the following:

- You change ```supports_distributed_data``` to a virtual member function ```supports_distributed_data``` to allow for (runtime) type polymorphism. I.e., the following is possible:
```
VectorBase *pointer = new VectorSpecialized;
pointer->supports_distributed_data();
```
I mean, this is in line with the abstract interfaces you defined for our linear algebra.

 - But, on the other hand you use it exclusively "type bound", i.e. your implementation above has the implicit assumption that creation of above ```VectorType``` is possible _and_ is a meaningful type. But what about a ```BlockVectorBase<VectorSpaceVector>```? [1]

A lot of this confusion is due to the fact that we happen to introduce a second form of polymorphism to an interface that formerly only had one. We now have:
 - Runtime type polymorphism due to inheritance from interface classes
 - Static type polymorphism due to templates

[1] The point is, we're already half way through transforming our linear algebra classes (at least for vectors) mainly to the first type of polymorphism---even though we use it mainly as a workaround for the lack of "concepts" in C++11/14/17. But by doing that we should support it completely, i.e., ```BlockVectorBase``` should ask its concrete block objects on runtime.